raiden/network/transport/matrix/client.py

```diff
@@ -603,12 +614,7 @@ def _handle_responses(
         for response in currently_queued_responses:
             for presence_update in response["presence"]["events"]:
                 for callback in list(self.presence_listeners.values()):
-                    self._worker_pool.spawn(
-                        callback, presence_update, next(self._presence_update_ids)
-                    )
-
-            # Collect finished greenlets and errors without blocking
-            self._worker_pool.join(timeout=0, raise_error=True)
+                    callback(presence_update, next(self._presence_update_ids))
```

Comment:
Quick note: We decided to remove the `spawn` here because of the sheer amount of presence updates. The global rooms have thousands of addresses and on the first sync (and at some points after in the Matrix sync loop) this would result in thousands of greenlets being spawn. This happened even though we used a poll, since the poll only limits the amount of greenlets that exist concurrently, but it still needs to spawn a new greenlet for each spawn call.

The greenlets were used here, because in the past the `sync` thread called the processing function directly, and for the few addresses that we actually wanted the presence updates we had to get the displayname and validate the signatures, which resulted in a timeout for the next `sync`. This is not a problem anymore since the processing of the Matrix response is done in a separate thread.

Also, this should be fast enough, since the vast majority of the addresses are not whitelisted, as a consequence the UserAdressManager will short-circuit the presence update. This will cost roughly two function calls and a dictionary lookup.